sympy/polys/domains/domain.py

```diff
@@ -278,6 +278,10 @@ def unify(K0, K1, symbols=None):
             else:
                 cls = K1.__class__
 
+            from sympy.polys.domains.old_polynomialring import GlobalPolynomialRing
```

Comment:
I wonder if for speed reasons, you should rather be importing this at the top of the module? I think importing each time can be slow.